crux/src/Crux.hs

```diff
@@ -144,44 +146,107 @@ postprocessSimResult opts res =
              cont (crux', os)
 
  `Ex.catch` \(e :: Ex.SomeException) ->
-   do let ?outputConfig = outCfg
+   do let ?outputConfig = mkOutCfg Nothing
       case (Ex.fromException e :: Maybe ExitCode) of
         Just exitCode -> exitWith exitCode
-        Nothing -> sayFail "Crux" (Ex.displayException e) >> exitFailure
+        Nothing -> logException e >> exitFailure
+
 
 showHelp :: Logs => Text -> Config opts -> IO ()
 showHelp nm cfg = do
   outWidth <- maybe 80 (\(Window _ w) -> w) <$> size
   let opts = LayoutOptions $ AvailablePerLine outWidth 0.98
-  outputLn (TL.unpack $ PR.renderLazy $ layoutPretty opts (configDocs nm cfg))
+  say OK "Crux" (PR.renderStrict $ layoutPretty opts (configDocs nm cfg))
 
 showVersion :: Logs => Text -> Version -> IO ()
 showVersion nm ver =
-  outputLn $ "crux version: " <> Crux.version <> ", " <> Text.unpack nm <> " version: " <> Version.showVersion ver
+  say OK "Crux" $ T.unwords [ "version: " <> T.pack Crux.version <> ","
+                            , nm
+                            , "version: " <> T.pack (Version.showVersion ver)
+                            ]
+
+
+-- | Create an OutputConfig for Crux, based on an indication of
+-- whether colored output should be used, the normal and error output
+-- handles, and the parsed CruxOptions.
+--
+-- If no CruxOptions are available (i.e. Nothing, as used in the
+-- preliminary portions of loadOptions), then a default output stance
+-- is applied; the default stance is described along with the
+-- individual option below.
+--
+-- The following CruxOptions affect the generated OutputConfig:
+--
+--  * printFailures  (default stance: True)
+--  * quietMode      (default stance: False)
+--  * simVerbose     (default stance: False)
+
+mkOutputConfig :: Bool -> Handle -> Handle -> Maybe CruxOptions -> OutputConfig
+mkOutputConfig withColors outHandle errHandle opts =
+  let lgWhat = let la = LJ.LogAction $ logToStd withColors outHandle errHandle
+                   -- TODO simVerbose may not be the best setting to use here...
+                   baseline = if maybe False ((> 1) . simVerbose) opts
+                              then Noisily
+                              else Simply
+               in if maybe False quietMode opts
+                  then logfltr OK >$< la
+                  else logfltr baseline >$< la
+      logfltr allowed = \case
+        SayNothing -> SayNothing
+        w@(SayWhat lvl _ _) -> if lvl >= allowed then w else SayNothing
+        SayMore m1 m2 -> case (logfltr allowed m1, logfltr allowed m2) of
+          (SayNothing, SayNothing) -> SayNothing
+          (SayNothing, m) -> m
+          (m, SayNothing) -> m
+          (m1', m2') -> SayMore m1' m2'
+
+      lgGoal = sayWhatFailedGoals (maybe True printFailures opts) >$< lgWhat
+      splitResults r@(CruxSimulationResult _cmpl gls) = (snd <$> gls, r)
+  in OutputConfig
+     {
+       _outputHandle = outHandle
+     , _quiet = maybe True quietMode opts
```

Comment:
Why `maybe True` here?

Reply:
Whoops, good catch, thanks!  Fixed in 946e1a7.